### PR TITLE
Add Rocky 9.2 repo versions

### PR DESCRIPTION
**Pull request by Alex-Welsh**
This change also fixes the formatting of the rocky CRB and highavailability repo versions so that the minor version is correctly chosen dynamically.

### CODE REVIEW

File: etc/kayobe/environments/ci-aio/stackhpc-ci.yml [hunk 1]
```diff
@@ -54,8 +54,8 @@ stackhpc_repo_centos_stream_9_docker_version: "{{ stackhpc_pulp_repo_centos_stre
 stackhpc_repo_rhel_9_treasuredata_4_version: "{{ stackhpc_pulp_repo_rhel_9_treasuredata_4_version }}"
 stackhpc_repo_rhel_9_mariadb_10_6_version: "{{ stackhpc_pulp_repo_rhel_9_mariadb_10_6_version }}"
 stackhpc_repo_rhel_9_influxdb_version: "{{ stackhpc_pulp_repo_rhel_9_influxdb_version }}"
-stackhpc_repo_rocky_9_crb_version: "{{ stackhpc_pulp_repo_rocky_9_1_crb_version }}"
-stackhpc_repo_rocky_9_highavailability_version: "{{ stackhpc_pulp_repo_rocky_9_1_highavailability_version }}"
+stackhpc_repo_rocky_9_crb_version: "{{ stackhpc_pulp_repo_rocky_9_crb_version }}"
```
Comment (priteau) [inline etc/kayobe/environments/ci-aio/stackhpc-ci.yml]: I think you need to replace all `9` by `9_2` in the references to variables.

Comment (m-bull) [reply etc/kayobe/environments/ci-aio/stackhpc-ci.yml]: Also this guy: https://github.com/stackhpc/stackhpc-kayobe-config/blob/9e6ffaa949618ff129aceca92498f80cb1879b7e/etc/kayobe/pulp.yml#L216

Comment (Alex-Welsh) [reply etc/kayobe/environments/ci-aio/stackhpc-ci.yml]: The minor version does need to change, but I think the 9 reference should all be templated [here](https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/yoga/etc/kayobe/pulp.yml#LL219C5-L219C5)

Comment (Alex-Welsh) [reply etc/kayobe/environments/ci-aio/stackhpc-ci.yml]: I've updated the minor version and added a reno. I'm a little concerned about updating the default minor version but 1) we don't have many Rocky 9 deployments yet and 2) I'm assuming it won't cause any major issues

Comment (priteau) [reply etc/kayobe/environments/ci-aio/stackhpc-ci.yml]: I had missed that `stackhpc_pulp_repo_rocky_9_crb_version` (for example) was defined elsewhere.

**Review by priteau — DISMISSED** [review]

File: etc/kayobe/environments/ci-multinode/stackhpc-ci.yml [hunk 2]
```diff
@@ -51,8 +51,8 @@ stackhpc_repo_centos_stream_9_docker_version: "{{ stackhpc_pulp_repo_centos_stre
 stackhpc_repo_rhel_9_treasuredata_4_version: "{{ stackhpc_pulp_repo_rhel_9_treasuredata_4_version }}"
 stackhpc_repo_rhel_9_mariadb_10_6_version: "{{ stackhpc_pulp_repo_rhel_9_mariadb_10_6_version }}"
 stackhpc_repo_rhel_9_influxdb_version: "{{ stackhpc_pulp_repo_rhel_9_influxdb_version }}"
-stackhpc_repo_rocky_9_crb_version: "{{ stackhpc_pulp_repo_rocky_9_1_crb_version }}"
-stackhpc_repo_rocky_9_highavailability_version: "{{ stackhpc_pulp_repo_rocky_9_1_highavailability_version }}"
+stackhpc_repo_rocky_9_crb_version: "{{ stackhpc_pulp_repo_rocky_9_crb_version }}"
```
Comment (Alex-Welsh) [inline etc/kayobe/environments/ci-multinode/stackhpc-ci.yml]: I've just noticed I should have moved these lower down below the comment. I'll update them once the sync is finished, which will conveniently re-run CI

**Comment by Alex-Welsh** [comment]
Promotion is done: https://github.com/stackhpc/stackhpc-release-train/actions/runs/5253811217

**Review by priteau — DISMISSED** [review]
It works fine, thanks @Alex-Welsh.

**Comment by markgoddard** [comment]
RL9 jobs failing:

```
TASK [dev-tools : Ensure required packages are installed] **********************
Monday 19 June 2023  15:11:49 +0000 (0:00:01.790)       0:01:29.760 *********** 
fatal: [controller0]: FAILED! => changed=false 
  msg: 'Failed to download metadata for repo ''extras'': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried'
  rc: 1
  results: []
```

**Comment by Alex-Welsh** [comment]
> RL9 jobs failing

`stackhpc_pulp_repo_rocky_9_minor_version` now set at 1 for CI

We should be using workflow-built images in CI soon and it'll be reverted then

**Comment by markgoddard** [comment]
> > RL9 jobs failing
> 
> `stackhpc_pulp_repo_rocky_9_minor_version` now set at 1 for CI
> 
> We should be using workflow-built images in CI soon and it'll be reverted then

Could you explain the problem?

**Comment by Alex-Welsh** [comment]
> > > RL9 jobs failing
> > 
> > 
> > `stackhpc_pulp_repo_rocky_9_minor_version` now set at 1 for CI
> > We should be using workflow-built images in CI soon and it'll be reverted then
> 
> Could you explain the problem?

My assumption is that because the default minor version is now 2, it's trying to use Rocky 9.2 repos when the CI image is fixed at 9.1 for now, which is causing it to fail

**Comment by markgoddard** [comment]
> > > > RL9 jobs failing
> > > 
> > > 
> > > `stackhpc_pulp_repo_rocky_9_minor_version` now set at 1 for CI
> > > We should be using workflow-built images in CI soon and it'll be reverted then
> > 
> > 
> > Could you explain the problem?
> 
> My assumption is that because the default minor version is now 2, it's trying to use Rocky 9.2 repos when the CI image is fixed at 9.1 for now, which is causing it to fail

Shouldn't be an issue if we overwrite all of the repos to 9.2. Perhaps we're missing a repo?

**Comment by m-bull** [comment]
Some of those repo versions exist on Ark, but not on test-pulp because of failed syncs. The failed syncs are (mostly) fixed now.

**Comment by Alex-Welsh** [comment]
> Some of those repo versions exist on Ark, but not on test-pulp because of failed syncs. The failed syncs are (mostly) fixed now.

Seems to be working now:  [OVS](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/5347456627/jobs/9696192612#step:17:343), [OVN](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/5347456627/jobs/9696193374#step:17:343) 
Broken ones for comparison: [OVS](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/5290672090/jobs/9692056700#step:17:343), [OVN](https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/5290672090/jobs/9692056496#step:17:343)

**Review by markgoddard — APPROVED** [review]
We need to get this RL CI sorted as we're a bti blind at the moment.

**Pull request merged**